src/system_descriptor.jl

```diff
@@ -9,27 +9,24 @@ function SystemArgument(;
     default,
     allowed_values,
 )
-    return SystemArgument(
-        name,
-        default,
-        allowed_values,
-    )
-end
-
-function get_allowed_values(arg::SystemArgument)
-    if isempty(arg.allowed_values)
-        error("allowed values are not defined for $arg")
+    if isempty(allowed_values)
```

Comment:
This should go into the inner constructor. Otherwise, someone could create an invalid instance by using the default constructor.

```
struct SystemArgument
     name::Symbol
     default::Any
     allowed_values::Set{<:Any}

    function SystemArgument(name, default, allowed_values)
        isempty(allowed_values) && error(“allowed_values cannot be empty”)
        new(name, default, allowed_values)
    end
end
```